chia/types/blockchain_format/program.py

```diff
@@ -148,8 +148,8 @@ class SerializedProgram:
 
     @classmethod
     def parse(cls, f) -> "SerializedProgram":
-        tmp = sexp_buffer_from_stream(f)
-        return SerializedProgram.from_bytes(tmp)
+        length = serialized_length(f.getvalue()[f.tell() :])
```

Comment:
It's possible that casting the bytes into a `memoryview` will save one copy of the blob. Might be worth checking to see how it benchmarks.

```python
length = serialized_length(memoryview(f.getvalue())[f.tell():])
```

(I've been meaning to experiment with `memoryview` in python `clvm` to see if it makes for zero-cost `substr`.)

Reply:
yeah, my initial idea was to use `getbuffer()`, but to feel confident with that API I would want to add more tests to the actual python bindings of `clvm_rs` too.

I agree that a longer term solution would be to take buffer views.